torch_geometric/explain/algorithm/pgm_explainer.py

```diff
@@ -0,0 +1,440 @@
+            pgm_nodes, p_values, candidate_nodes = self._explain_graph(
+                model=model, x=x, index=index, target=target,
+                num_samples=num_samples, top_node=top_node,
+                significance_threshold=significance_threshold,
+                pred_threshold=pred_threshold, **kwargs)
+            return Explanation(node_mask=p_values < significance_threshold)
+
+        # if explainer_config.node_mask_type == MaskType.object:
+        #     node_feat_mask = None
+        # else:
+        #     node_feat_mask = node_mask
+        #     node_mask = None
+
+        # build explanation
+        return Explanation(
+            x=x,
+            edge_index=edge_index,
+            markov_blanket=neighbors,
+            # pgm_explation=pgm_explanation,
+            # marginal_probability=marginal_prob,
+            pgm_stats=pgm_stats,
+        )
+
+    def batch_perturb_features_on_node(
+            self,
+            num_samples,
+            index_to_perturb,
+            X_features,
+            model,
+            percentage=50,  # % time node gets pertubed
+            p_threshold=0.05,
+            pred_threshold=0.1,
+            pertubation_mode="mean",
+            snorm_node=None,
+            snorm_edge=None,
+            **kwargs):
+        # for pertubing a batch of graphs for graph classification tasks
+
+        pred_torch = model(X_features, **kwargs)
+        soft_pred = torch.softmax(pred_torch, dim=1)
+        pred_label = torch.argmax(soft_pred, dim=1)
+        num_nodes = X_features.shape[0]
+
+        samples = []
+        for iteration in range(num_samples):
+            X_perturb = X_features.detach().clone()
+            sample = []
+            for node in range(num_nodes):
+                if node in index_to_perturb:
+                    seed = np.random.randint(100)
+                    if seed < percentage:
+                        latent = 1
+                        X_perturb = self.perturb_features_on_node(
+                            X_perturb, node, pertubation_mode=pertubation_mode,
+                            perturb_feature_list=self.perturb_feature_list,
+                            is_random=True)
+                    else:
+                        latent = 0
+                else:
+                    latent = 0
+                sample.append(latent)
+            # if data:
+            #     data.x = X_perturb.detach().clone()
+            # kwargs['data'] = data
+            pred_perturb_torch = model(X_perturb, **kwargs)
+            soft_pred_perturb = torch.softmax(pred_perturb_torch,
+                                              dim=1).squeeze()
+
+            pred_change = torch.max(soft_pred) - soft_pred_perturb[pred_label]
+
+            sample.append(pred_change)
+            samples.append(sample)
+
+        samples = torch.tensor(samples)
+        if self.perturb_indicator == "abs":
+            samples = torch.abs(samples)
+
+        top = int(num_samples / 8)
```

Comment:
Can we use a user argument to replace 8 here? The user can determine the number of top samples.

Reply:
I haven't changed this yet-- I am still a little bit unsure what the `top` variable is doing (from https://github.com/vunhatminh/PGMExplainer/blob/715402aa9a014403815f518c4c7d9258eb49bbe9/PGM_Graph/pgm_explainer_graph.py#L112-L118 , or their choice of dividing by 8 -- it seems to be overwriting the `pred_change` variable that they calculated above